lib/oneview-sdk/image-streamer/client.rb

```diff
@@ -0,0 +1,154 @@
+        raise InvalidClient, 'Must set the url option' unless @url
+        @max_api_version = appliance_api_version
+        if options[:api_version] && options[:api_version].to_i > @max_api_version
+          logger.warn "API version #{options[:api_version]} is greater than the Image Streamer API version (#{@max_api_version})"
+        end
+        @api_version = options[:api_version] || [OneviewSDK::ImageStreamer::DEFAULT_API_VERSION, @max_api_version].min
+        # Set the default Image Streamer module API version
+        OneviewSDK::ImageStreamer.api_version = @api_version unless
+          OneviewSDK::ImageStreamer.api_version_updated? || !OneviewSDK::ImageStreamer::SUPPORTED_API_VERSIONS.include?(@api_version)
+        @ssl_enabled = true
+        if ENV.key?('ONEVIEWSDK_I3S_SSL_ENABLED')
+          if %w(true false 1 0).include?(ENV['ONEVIEWSDK_I3S_SSL_ENABLED'])
+            @ssl_enabled = !%w(false 0).include?(ENV['ONEVIEWSDK_I3S_SSL_ENABLED'])
+          else
+            @logger.warn "Unrecognized ssl_enabled value '#{ENV['ONEVIEWSDK_I3S_SSL_ENABLED']}'. Valid options are 'true' & 'false'"
+          end
+        end
+        @ssl_enabled = options[:ssl_enabled] unless options[:ssl_enabled].nil?
+        @timeout = options[:timeout] unless options[:timeout].nil?
+        @cert_store = OneviewSDK::SSLHelper.load_trusted_certs if @ssl_enabled
+        raise InvalidClient, 'Must set token option' unless options[:token] || ENV['ONEVIEWSDK_I3S_TOKEN']
+        @token = options[:token] || ENV['ONEVIEWSDK_I3S_TOKEN']
+      end
+
+      # Tells OneView to create the resource using the current attribute data
+      # @param [Resource] resource the object to create
+      def create(resource)
+        resource.client = self
+        resource.create
+      end
+
+      # Sets the attribute data, and then saves to OneView
+      # @param [Resource] resource the object to update
+      def update(resource, attributes = {})
+        resource.client = self
+        resource.update(attributes)
+      end
+
+      # Updates this object using the data that exists on OneView
+      # @param [Resource] resource the object to refresh
+      def refresh(resource)
+        resource.client = self
+        resource.refresh
+      end
+
+      # Deletes this object from OneView
+      # @param [Resource] resource the object to delete
+      def delete(resource)
+        resource.client = self
+        resource.delete
+      end
+
+      # Get array of all resources of a specified type
+      # @param [String] type Resource type
+      # @param [Integer] api_ver API module version to fetch resources from
+      # @return [Array<Resource>] Results
+      # @example Get all Deployment Plans
+      #   deployment_plans = @client.get_all('DeploymentPlans')
+      # @raise [TypeError] if the type is invalid
+      def get_all(type, api_ver = @api_version)
+        klass = OneviewSDK.resource_named(type, api_ver)
```

Comment:
We'll need to make a new `OneviewSDK::ImageStreamer.resource_named` method and call that here, won't we? The previous one, defined [here](https://github.com/HewlettPackard/oneview-sdk-ruby/blob/master/lib/oneview-sdk/resource.rb#L354) only looks through the OneView resources.